packages/measure/lib/commands/ioscpugpu/new.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:measure/commands/base.dart';
+import 'package:measure/parser.dart';
+
+class IosCpuGpuNew extends IosCpuGpuSubcommand {
+  IosCpuGpuNew() {
+    argParser.addOption(
+      kOptionTimeLimitMs,
+      abbr: 'l',
+      defaultsTo: '5000',
+      help: 'time limit (in ms) to run instruments for measuring',
+    );
+    argParser.addOption(
+      kOptionTemplate,
+      abbr: 't',
+      help: 'instruments template'
+    );
+    argParser.addOption(
+      kOptionDevice,
+      abbr: 'w',
+      help: 'device identifier recognizable by instruments '
+            '(e.g., 00008020-000364CE0AF8003A)',
+    );
+  }
+
+  @override
+  String get name => 'new';
+  @override
+  String get description => 'Take a new measurement on the iOS CPU/GPU '
+                            'percentage (of Flutter Runner).';
+
+  String get _timeLimit => argResults[kOptionTimeLimitMs];
+
+  @override
+  Future<void> run() async {
+    checkRequiredOption(kOptionTemplate);
+    checkRequiredOption(kOptionTraceUtility);
+    _checkDevice();
+
+    print('Running instruments on iOS device $_device for ${_timeLimit}ms');
+
+    final List<String> args = <String>[
+      '-l', _timeLimit,
+      '-t', argResults[kOptionTemplate],
+      '-w', _device,
+    ];
+    if (isVerbose) {
+      print('instruments args: $args');
+    }
+    final ProcessResult processResult = Process.runSync('instruments', args);
+    _parseTraceFilename(processResult.stdout.toString());
+
+    print('Parsing $_traceFilename');
+
+    final CpuGpuResult result =
+        IosTraceParser(isVerbose, traceUtility).parseCpuGpu(_traceFilename, processName);
+    result.writeToJsonFile(outJson);
+    print('$result\nThe result has been written into $outJson');
+  }
+  String _traceFilename;
+  Future<void> _parseTraceFilename(String out) async {
+    const String kPrefix = 'Instruments Trace Complete: ';
+    final int prefixIndex = out.indexOf(kPrefix);
+    if (prefixIndex == -1) {
+      throw Exception('Failed to parse instruments output:\n$out');
+    }
+    _traceFilename = out.substring(prefixIndex + kPrefix.length).trim();
+  }
+
+  String _device;
+  void _checkDevice() {
+    _device = argResults[kOptionDevice];
+    if (_device == null) {
+      final ProcessResult result = Process.runSync('flutter', <String>['devices']);
```

Comment:
GitHub lost some comments :(

I think we should not do this. Flutter tool output is not guarnateed to be stable, and an upstream change could easily break this unintentionally. I would recommend making the device argument be required, and callers could use flutter devices or whatever other method to call it (in the devicelab, we know the device ID we're supposed to target anyway).

Reply:
I've already let the devicelab provide the deviceId in https://github.com/flutter/flutter/pull/39439

However, I'd like my local use and test to be a little easier (see the last test in measure_test.dart).

I've now removed the `flutter` and instead used `instruments` to find the deviceId.